### ISSUE TITLE
fix packaging

### ISSUE THREAD

**Issue by NickleDave** [yardencsGitHub/tweetynet#177]
the current source code layout that we've evolved into since 0.6.0 release + setup.py causes a couple of issues:
- the `article` package that I added in `src` is getting added to the source distribution, because the `setup.py` file says "just include every package you find in the `src/` directory"
- this causes a bunch of import errors -, if you don't do `pip install .[article]` to get dependencies like `sklearn` that are declared in the `EXTRAS_REQUIRE` of the `setup.py`
  + -> to reproduce, do `$ python setup.py sdist` and then `pip install` the generated tar.gz in the `dist/` directory, then launch a REPL and try to `import tweetynet`
- the intent of the `article` package is not to be in the `tweetynet` distribution -- it just gives us a separate convenient package to work with for article-only code
- but the `yarden2annot` module is currently in that `article` package. `vak` needs access to it, to load the canary annotations.
- also this is all extra annoying to figure out how to fix, because we're still using an old-school `setup.py` which is not even well-documented any more on the `setuptools` site 

fix:
- [ ] adopt a "monorepo" format where we actually have two sub-dirs (see below)
- [ ] switch to declarative setup.cfg instead of setup.py, or possibly switch to `poetry` and use `pyproject.toml`
- [ ] move `yarden2annot` into `tweetynet` package
- [ ] fix corresponding entry point in tweetynet/setup.cfg file upon doing so, so the `format` still gets installed for `vak`
- [ ] add install instructions for `article` code -- we shouldn't put `article` on PyPI but we should make it easy for an interested reader to install locally. Have a separate README in the "article" sub-directory for this

This is the structure I mean by "monorepo",
I think this is the least-complicated solution. An alternative might be doing something fancy with setup.cfg and/or setup.py but I'm not sure if it's possible and if so worth it

```
tweetynet 
│   ├── tweetynet
│   │   ├── src
│   │   ├── setup.cfg 
│   │   └── tests/
│   └── article
│   │   ├── src
│   │   ├── setup.cfg 
```

**Comment by NickleDave** [comment]
Also, these issues are what's preventing me from testing whether the fix for newer versions of `torch` works. Hence the need to fix sooner rather than later

**Comment by NickleDave** [comment]
I ended up solving this differently.
- [ ] remove `yarden2annot` and added that directly to `crowsetta`, released a new version of `crowsetta`; because I was running into issues with circular imports and the other workarounds I came up with would also have been weird and fragile. Also because we want it to be easy for people to be able to work with those annotations, without needing to install tweetynet necessarily
- [ ] keep the `tweetynet/src` repo at the same level -- this is the "package" that this repo can be used to build
- [ ] add the `article` folder and just move all related things into that folder. It will be easy to install but it's not set up to build into a distribution package, which is fine
- [ ] keep pyproject.toml but use  `flit` to develop -- the pyproject.toml is the standard used by more projects and I find it much more readable than setup.cfg so I would rather use it
